cmake/modules/AddSwift.cmake

```diff
@@ -856,6 +850,31 @@ function(_add_swift_library_single target name)
     set(SWIFTLIB_SINGLE_API_NOTES "${module_name}")
   endif()
 
+  # On Linux and FreeBSD add conformance magic objects that coalesce protocol
+  # conformance sections. This wouldn't be necessary if the link was done by
+  # the swift binary: rdar://problem/19007002
+  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
+     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+
+    if("${libkind}" STREQUAL "SHARED")
+      set(arch_subdir "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}")
+
+      add_custom_command_target(conformance_object_begin
```

Comment:
I think that would hurt the generality of this approach.  Because add_swift_library_single accepts a triple, it should be assumed that the triple could change between invocations.  If this logic was factored out of the function, it's possible that the conformance objects from one target would be linked against a shared library from another.